crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -374,6 +375,122 @@ impl Persistence {
 
         Ok(solution)
     }
+
+    /// Returns the oldest settlement event for which the accociated auction is
+    /// not yet populated in the database.
+    pub async fn get_settlement_without_auction(
+        &self,
+    ) -> Result<Option<domain::settlement::Event>, error::SettlementEvent> {
+        let _timer = Metrics::get()
+            .database_queries
+            .with_label_values(&["get_settlement_without_auction"])
+            .start_timer();
+
+        let mut ex = self.postgres.pool.begin().await.map_err(DatabaseError)?;
+        let event = database::settlements::get_settlement_without_auction(&mut ex)
+            .await
+            .map_err(DatabaseError)?
+            .map(|event| {
+                let event = settlement::Event {
+                    inner: eth::Event {
+                        block: u64::try_from(event.block_number)
+                            .map_err(|_| error::SettlementEvent::InvalidEvent)?
+                            .into(),
+                        log_index: u64::try_from(event.log_index)
+                            .map_err(|_| error::SettlementEvent::InvalidEvent)?,
+                    },
+                    transaction: eth::TxId(H256(event.tx_hash.0)),
+                };
+                Ok::<_, error::SettlementEvent>(event)
+            })
+            .transpose()?;
+        Ok(event)
+    }
+
+    pub async fn save_settlement(
+        &self,
+        event: domain::settlement::Event,
+        auction_id: domain::auction::Id,
+        settlement: Option<&domain::settlement::Settlement>,
```

Comment:
Bit unclear from just reading the function signature under which circumstances it makes sense to have `settlement` be `None`.
I assume this happens in the case where we index a settlement event coming from the other environment so we are not able to find the necessary data in the DB?
In that case does it not make sense to split this function into 2?
One for handling just the event and one for storing the settlement information (in case there is some).

Reply:
> In that case does it not make sense to split this function into 2

Probably, but then we would have two `commit` statements, for which we would need to significantly change how the OnSettlementEventUpdater works. It would have to be split in two updates, one update for <auction, settlement> relation and other update that would have to check maybe settlement_observation table if it misses an entry for any of the settlement events etc. Anyway, I think this should be done but outside of scope of this PR.

Reply:
I don't have the full context back in my head but is passing the transaction to the second function not an option to get around the 2-commit issue?